sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala

```diff
@@ -885,6 +936,12 @@ class LongHashedRelation(
    * Returns an iterator for keys of InternalRow type.
    */
   override def keys(): Iterator[InternalRow] = map.keys()
+
+  override def values(): Iterator[InternalRow] = {
```

Comment:
I am a bit confused about the meaning of 'value' : Is it the "value" of the hash-table (if we think of a hash-table as a dictionary or a key-value store), or is it the "extra bit at the end to know if the row was matched or not" ?

I believe the use of the word 'value' is a bit overloaded in this PR. Personally, I think it would be great if you don't use the word value for the second concept of "extra bit at the end to know if the row was matched or not".

Reply:
@agrawaldevesh - "value" here means the value of the hash-table. Actually I don't use `value` for extra bit at the end of row in this PR. I use `value` either to be expressions of original value before creating a hash-table (`UnsafeHasedRelation.apply()`), or here to be the values of hash-table.